src/lib.rs

```diff
@@ -153,17 +153,20 @@ fn init_logger(config: &Config) {
         .with_writer(std::io::stderr)
         .with_env_filter(filter);
 
-    if config.human_logging {
+    let inited = if config.human_logging {
         // use pretty logs
-        sub.init();
+        sub.try_init()
     } else {
         // use machine parseable structured logs
         sub
             // disable terminal colors
             .with_ansi(false)
             // use json
             .json()
-            .init();
+            .try_init()
+    };
+    if let Err(err) = inited {
```

Comment:
If we can't init the logger we shouldn't start the service imo

Reply:
In unit tests we can't init it two timtes=(

Reply:
Hmm we should move the logger init to a higher level fn than `start_service`, and only call it from main then.